notification-server/src/processor.rs

```diff
@@ -101,3 +102,294 @@ pub async fn process(
+                    }),
+                },
+                BlockItemSummary {
+                    index: random_transaction_index(),
+                    energy_cost: ArbitraryEnergy::arbitrary(g).0,
+                    hash: fixed_hash(),
+                    details: BlockItemSummaryDetails::AccountTransaction(account_transaction_details(AccountTransactionEffects::None {
+                        reject_reason: RejectReason::OutOfEnergy,
+                        transaction_type: Some(TransactionType::Update),
+                    })),
+                },
+                BlockItemSummary {
+                    index: random_transaction_index(),
+                    energy_cost: ArbitraryEnergy::arbitrary(g).0,
+                    hash: fixed_hash(),
+                    details: BlockItemSummaryDetails::AccountTransaction(account_transaction_details(AccountTransactionEffects::TransferredToEncrypted {
+                        data: Box::new(EncryptedSelfAmountAddedEvent {
+                            amount: amount.clone(),
+                            account: receiver_address.clone(),
+                            new_amount: EncryptedAmount {
+                                encryptions: [get_random_cipher(), get_random_cipher()],
+                            },
+                        }),
+                    })),
+                }
+            ];
+
+            SilentBlockItemSummary(g.choose(&silent_block_summaries).unwrap().clone())
+        }
+    }
+    #[derive(Clone, Debug)]
+    enum BlockItemSummaryWrapper {
+        Emitting(EmittingBlockItemSummaryPair),
+        Silent(SilentBlockItemSummary),
+    }
+
+    impl Arbitrary for BlockItemSummaryWrapper {
+        fn arbitrary(g: &mut Gen) -> Self {
+            if bool::arbitrary(g) {
+                BlockItemSummaryWrapper::Emitting(EmittingBlockItemSummaryPair::arbitrary(g))
+            } else {
+                BlockItemSummaryWrapper::Silent(SilentBlockItemSummary::arbitrary(g))
+            }
+        }
+    }
+
+    trait TestableBlockItemSummary {
+        fn get_block_item_summary(&self) -> &BlockItemSummary;
+        fn get_expected_notification(&self) -> Option<NotificationInformation>;
+    }
+
+    impl TestableBlockItemSummary for BlockItemSummaryWrapper {
+        fn get_block_item_summary(&self) -> &BlockItemSummary {
+            match self {
+                BlockItemSummaryWrapper::Emitting(pair) => &pair.0,
+                BlockItemSummaryWrapper::Silent(summary) => &summary.0,
+            }
+        }
+
+        fn get_expected_notification(&self) -> Option<NotificationInformation> {
+            match self {
+                BlockItemSummaryWrapper::Emitting(pair) => Some(pair.1.clone()),
+                BlockItemSummaryWrapper::Silent(_) => None,
+            }
+        }
+    }
+
+    #[quickcheck]
+    fn test_random_block_item_summary(summaries: Vec<BlockItemSummaryWrapper>) -> bool {
```

Comment:
I'm not too familiar with prop testing in general but on my machine the test was running for more than 60 seconds. Is it possible to configure the amount of arbitrary runs done?

Reply:
We should be cautious with hardcoding limits in proptesting. It tests the boundaries based on some reductions. If it does not test those boundaries according to its definition, they might be less effective.

In here we have a lot of different branches, so the total amount of boundaries is immense. 

Around a minute of test time is not a worry of mine.